addons/web_editor/controllers/main.py

```diff
@@ -626,6 +707,23 @@ def modify_image(self, attachment, res_model=None, res_id=None, name=None, data=
                 url_fragments = attachment.url.split('/')
                 url_fragments.insert(-1, str(attachment.id))
                 attachment.url = '/'.join(url_fragments)
+
+        # Update the image.data record
+        image_data_res_info = self._get_image_data_res_info(res_model=res_model, res_id=res_id, res_field=res_field, res_type=res_type, new_attachment_id=attachment.id)
+        image_data_res_model = image_data_res_info['image_data_res_model']
+        image_data_res_field = image_data_res_info['image_data_res_field']
+        image_data_res_id = image_data_res_info['image_data_res_id']
+        image_data = request.env['web_editor.image.data'].search([
+            ('res_model', '=', image_data_res_model),
+            ('res_id', '=', image_data_res_id),
+            ('res_field', '=', image_data_res_field),
```

Comment:
@qsm-odoo,  I was actually hesitating to put it but as there is `check_unique_record()` in the new model (that ensures that there is not more than one record per res_model-res_id-res_field), I thought, it would maybe be considered as redundant? (I am not sure of what it is the best thing to do in this case)

Reply:
- I am pretty sure that adding `limit=1` would improve perf whatever the unique constraints are (but of course, the unique constraints is more important than this `limit=1`).
- Even if I am wrong, from a readability point of view, adding `limit=1` is good: reading the code it clearly indicates that you are expecting one result.